packages/graphback-runtime-knex/tests/data/KnexDbDataProviderTest.ts

```diff
@@ -4,7 +4,8 @@
 import { unlinkSync, existsSync } from 'fs';
 import { buildSchema } from 'graphql';
 import * as Knex from 'knex';
-import { GraphbackDataProvider, GraphbackCoreMetadata, QueryFilter } from '@graphback/core';
+import { GraphbackDataProvider, GraphbackPluginEngine, QueryFilter } from '@graphback/core';
+import { SchemaCRUDPlugin } from '../../../graphback-codegen-schema/src';
```

Comment:
can we make `graphback-codegen-schema` dev dependency and use that here instead of relative path?

Reply:
I tried importing the actual package, but got this error, which is isolated to the test files, not sure why it is happening

```shell
Type 'SchemaCRUDPlugin' is not assignable to type 'GraphbackPlugin'.
  Types of property 'transformSchema' are incompatible.
    Type '(metadata: import("/home/ephelan/code/aerogear/graphback/packages/graphback-runtime-knex/node_modules/@graphback/codegen-schema/node_modules/@graphback/core/types/plugin/GraphbackCoreMetadata").GraphbackCoreMetadata) => import("/home/ephelan/code/aerogear/graphback/node_modules/graphql/type/schema").GraphQLSchema' is not assignable to type '(metadata: import("/home/ephelan/code/aerogear/graphback/packages/graphback-core/types/plugin/GraphbackCoreMetadata").GraphbackCoreMetadata) => import("/home/ephelan/code/aerogear/graphback/node_modules/graphql/type/schema").GraphQLSchema'.
      Types of parameters 'metadata' and 'metadata' are incompatible.
        Type 'import("/home/ephelan/code/aerogear/graphback/packages/graphback-core/types/plugin/GraphbackCoreMetadata").GraphbackCoreMetadata' is not assignable to type 'import("/home/ephelan/code/aerogear/graphback/packages/graphback-runtime-knex/node_modules/@graphback/codegen-schema/node_modules/@graphback/core/types/plugin/GraphbackCoreMetadata").GraphbackCoreMetadata'.
          Types have separate declarations of a private property 'supportedCrudMethods'.
```

Reply:
Ah I would not expect that error. Anyway, we can ignore this comments to not derail us from getting this in. 

Maybe we can open a separate issue for this?

Reply:
Yeah, and when I added it as a devDependency I got a circular dependency warning, so that is possibly why:

```
lerna WARN ECYCLE @graphback/codegen-schema -> @graphback/core -> @graphback/codegen-schema
```

Reply:
Okay, that explains it.